tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementMainPageViewModel.cs

```diff
@@ -85,38 +175,192 @@ public void HideRepository(RepositoryManagementItemViewModel repository)
+        }
+
+        return filteredAndSortedRepositories.ToList();
+    }
+
+    private string GetConfigurationFileIfExists(string repositoryPath)
+    {
+        var configurationDirectory = Path.Join(repositoryPath, DscHelpers.ConfigurationFolderName);
+        if (Directory.Exists(configurationDirectory))
+        {
+            var fileToUse = Directory.EnumerateFiles(configurationDirectory)
+            .Where(file => file.EndsWith(DscHelpers.ConfigurationFileYamlExtension, StringComparison.OrdinalIgnoreCase) ||
+                           file.EndsWith(DscHelpers.ConfigurationFileWingetExtension, StringComparison.OrdinalIgnoreCase))
+            .OrderByDescending(configurationFile => File.GetLastWriteTime(configurationFile))
+            .FirstOrDefault();
+
+            if (fileToUse != null)
+            {
+                return fileToUse;
+            }
+        }
+
+        return string.Empty;
+    }
+
+    private List<(Repository, Commit)> AssignCommitToRepository(List<Repository> repositories)
+    {
+        var repositoriesButWithCommits = new List<(Repository, Commit)>();
+
+        foreach (var repository in repositories)
+        {
+            var sourceControlId = repository.HasAssignedSourceControlProvider ? repository.SourceControlClassId ?? Guid.Empty : Guid.Empty;
+            var latestCommit = GetLatestCommitInformation(repository.RepositoryClonePath, sourceControlId);
+
+            repositoriesButWithCommits.Add((repository, latestCommit));
+        }
+
+        return repositoriesButWithCommits;
+    }
+
+    private Commit GetLatestCommitInformation(string repositoryLocation, Guid sourceControlProviderClassId)
+    {
+        if (sourceControlProviderClassId == Guid.Empty)
+        {
+            _log.Warning($"sourceControlProviderClassId is guid.empty. {sourceControlProviderClassId}.  Can not get commit information");
+            return Commit.DefaultCommit;
+        }
+
+        // This call does check the settings for file explorer and souce control integration.
+        var repositoryProperties = _enhanceRepositoryService.GetProperties(_commitProperties, repositoryLocation);
+
+        var isAPropertyMissing = false;
+        if (!repositoryProperties.TryGetValue("System.VersionControl.LastChangeAuthorName", out var latestCommitAuthorName))
+        {
+            latestCommitAuthorName = string.Empty;
+            isAPropertyMissing = true;
+        }
+
+        if (!repositoryProperties.TryGetValue("System.VersionControl.LastChangeDate", out var latestCommitChangedDate))
+        {
+            latestCommitChangedDate = DateTime.UnixEpoch;
+            isAPropertyMissing = true;
+        }
+
+        if (!repositoryProperties.TryGetValue("System.VersionControl.LastChangeID", out var latestCommitSHA))
+        {
+            latestCommitSHA = string.Empty;
+            isAPropertyMissing = true;
+        }
```

Comment:
can't all of this just be

    if (!repositoryProperties.TryGetValue("System.VersionControl.LastChangeAuthorName", out var latestCommitAuthorName))
    {
        return Commit.DefaultCommit;
    }

    if (!repositoryProperties.TryGetValue("System.VersionControl.LastChangeDate", out var latestCommitChangedDate))
    {
        return Commit.DefaultCommit;
    }

    if (!repositoryProperties.TryGetValue("System.VersionControl.LastChangeID", out var latestCommitSHA))
    {
        return Commit.DefaultCommit;
    }

?

Reply:
Only a smart person would figure this out.

Nice catch.  Changed.